Sources/Controllers/Cells/RouteInfoListItemCell.swift

```diff
@@ -0,0 +1,84 @@
+//
+//  RouteInfoListItemCell.swift
+//  OsmAnd Maps
+//
+//  Created by Max Kojin on 27/05/24.
+//  Copyright © 2024 OsmAnd. All rights reserved.
+//
+
+import UIKit
+
+@objc(OARouteInfoListItemCell)
+@objcMembers
+final class RouteInfoListItemCell: UITableViewCell {
+    
+    @IBOutlet private weak var leftImageView: UIImageView!
+    @IBOutlet private weak var topLeftLabel: UILabel!
+    @IBOutlet private weak var topRightLabel: UILabel!
+    @IBOutlet private weak var bottomLabel: UILabel!
+    @IBOutlet private weak var bottomImageView: UIImageView!
+    @IBOutlet private weak var bottomImageBorderView: UIView!
+    @IBOutlet private weak var bottomImageBorderViewWidthConstraint: NSLayoutConstraint!
+    @IBOutlet private weak var bottomImageStackView: UIStackView!
+    
+    let bottomImageViewDefaultWidth: CGFloat = 30
+    let bottomImageViewInnerHeight: CGFloat = 22
+    
+    var leftTurnIconDrawable: OATurnDrawable?
+    
+    override class func getIdentifier() -> String {
+        "RouteInfoListItemCell"
+    }
+    
+    func setLeftImageView(image: UIImage?) {
+        leftImageView.image = image
+    }
+    
+    func setleftTurnIconDrawable(drawable: OATurnDrawable) {
+        leftTurnIconDrawable = drawable
+    }
+    
+    func setTopLeftLabel(text: String?) {
+        topLeftLabel.text = text
+    }
+    
+    func setTopRightLabel(text: String?) {
+        topRightLabel.text = text
+    }
+    
+    func setBottomLabel(text: String?) {
+        bottomLabel.text = text
+    }
+    
+    func setBottomLanesImage(image: UIImage?) {
+        if let image {
+            bottomImageStackView.isHidden = false
+            bottomImageView.image = image
+            
+            bottomImageBorderView.layer.cornerRadius = 5
+            bottomImageBorderView.layer.borderWidth = 2
+            bottomImageBorderView.layer.borderColor = UIColor.iconColorDefault.cgColor
+            
+            bottomImageBorderViewWidthConstraint.constant = image.size.width / (image.size.height / bottomImageViewInnerHeight)
+            bottomImageBorderViewWidthConstraint.constant += 16
+        } else {
+            bottomImageStackView.isHidden = true
+        }
+    }
+    
+    override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        super.traitCollectionDidChange(previousTraitCollection)
+        if traitCollection.hasDifferentColorAppearance(comparedTo: previousTraitCollection) {
+            self.refreshLeftIcon()
+        }
+    }
+    
+    private func refreshLeftIcon() {
+        if let leftTurnIconDrawable {
+            leftTurnIconDrawable.clr = (traitCollection.userInterfaceStyle == .dark) ? .white : .black
```

Comment:
I don’t understand what do you mean and what I need to do? This code should redraw and recolor left icon with a turn lanes arrow image. Because it’s very custom and it doesn’t recolor by itself. This left image should be black on day mode and white on night mode.

Reply:
If you apply the default color .black, it will be black in the light theme and white in the dark one. Do you need a different behavior?